lib/spack/docs/environments.rst

```diff
@@ -0,0 +1,305 @@
+`Python Virtual Environments
+<https://docs.python.org/3/tutorial/venv.html>`_.  Spack environments
+some distinctive features:
+
+#. A spec installed "in" an environment is no different form the same
+   spec installed anywhere else in Spack.  Environments are assembled
+   simply by collecting together a set of specs.  Environments are
+   loaded by generating a set of ``module load`` commands.  (In the
+   future, environments can also be assembled into a single tree using
+   techniques from ``spack view``.)
+#. Spack Environments may contain more than one version of the same
+   package; but only a single module for a package may be loaded.
+   Modules that occur in earlier specs listed in an environment take
+   precedence over modules that occur later.
+
+Using Environments
+------------------
+
+Here we follow a typical use case of creating, concretizing,
+installing and loading an environment.
+
+.. note::
+
+   Environments are currently in Beta.  We expect the user interface
+   to change.  Future changes might also break existing environment
+   files, particularly ``environment.json``.
+
+
+Creating an Environment
+^^^^^^^^^^^^^^^^^^^^^^^
+
+An environment is created by:
+
+.. code-block:: console
+
+   $ spack env myenv create
+
+Spack then creates the following files:
+
+* The directory ``var/spack/environments/myenv``.  This is the public
+  part of the environment.
+
+  .. note::
+
+     All environments are stored in the ``var/spack/environments`` folder.
+
+  The following files may be added to this directory by the user or
+  Spack:
+
+  * ``env.yaml``: Addition environment specification and configuration
+  * ``loads``: A script of ``module load`` commands, generated by
+    ``spack env myenv loads``, which the user may source to use the
+    built environment.
+  * ``logs/``: Symbolic links to build logs for specs that are part of
+    the environment.
+  * Anything else the user wishes to put in this directory!
+
+* The directory ``var/spack/environments/myenv/.env``.  This directory
+  is "owned" by Spack, and may be rewritten from time to time.  The
+  user should not put anything in it.  It contains:
+
+  * ``environment.json``: Concretized specs for the environment.
+  * ``repo/``: A repo consisting of the Spack packages used in this
+    environment.  This allows the environment to build the same, in
+    theory, even on different verions of Spack with different
+    packages!
+
+Adding User Specs
+^^^^^^^^^^^^^^^^^
+
+Once an environment is created, it is time to add some user specs.  A
+*user spec* is simply a spec typed by the user; for example, what one
+would type as part of ``spack install``.  An environment can consist
+of as many user specs as one likes, and the specs can contain
+conflicting versions of packages.  For example:
+
+.. code-block:: console
+
+   $ spack env myenv add mpileaks
+   $ spack env myenv add python
+
+.. note::
+
+   Currently, only "bare" user specs may be added to an environment,
```

Comment:
@citibeth this was supposed to be fixed by https://github.com/spack/spack/pull/7843/commits/d537b69e3cb53c06f77e858b76e1fef4a19ee59d. Does it still not work for you?

Reply:
OK, I removed the text.